Source/RapyutaSimulationPlugins/Private/Core/RRGameSingleton.cpp

```diff
@@ -160,16 +166,74 @@ bool URRGameSingleton::InitializeResources(bool bInRequestResourceLoading)
     }
     else
     {
-        // NOTE: NO dynamic resources required -> considered to have been all loaded
-        for (auto& [_, resourceInfo] : ResourceMap)
+        // NOTE: NO initial resources loading required
+        // -> Considered to have been all loaded
+        for (auto& [dataType, resourceInfo] : ResourceMap)
         {
-            resourceInfo.bHasBeenAllLoaded = true;
+            resourceInfo.bHasBeenAllLoaded = (dataType != ERRResourceDataType::UE_DATA_TABLE);
+            // -> LoadObject<>() is expected to be called on [resourceInfo.GetAssetPath()] on-the-fly during Sim run
+            // -> Refer to [GetSimResource()]
         }
+        // Except [UE_DATA_TABLE], which should be always loaded initially
+        bResult &= RequestResourcesLoading<ERRResourceDataType::UE_DATA_TABLE>();
+
+        // -> Only collate assets' metadata without loading
+        CollateAssetsInfo();
+
+        // NOTE: Due to on-the-fly resource loading is only possible if running in Game-thread
+        // -> Preload global statically defined resources, in case they are referenced in non-game thread
+        // -> Could only load after collating assets info above
+        // StaticMeshes
+        GetStaticMesh(SHAPE_NAME_PLANE);
```

Comment:
Are these resources guaranteed to exist? In a packaged game, Unreal _should_ exclude anything that is not explicitly referenced in the scene hierarchy it generates to build all the packages.

Reply:
So `/RapyutaSimulationPlugins/DynamicContents` is expected to be added to `DirectoriesToAlwaysCook` in `DefaultGame.ini` like [here](https://github.com/rapyuta-robotics/flaptter_UE/blob/devel/Config/DefaultGame.ini#L86)
Otherwise, an error will be raised.
Since these are commonly used basic primitive shapes and master material used in our Sim, I keep them fetched here for ease of managing.
Regarding `DirectoriesToAlwaysCook`, I will add further guideline logs into the error, if user project does not have it configured.